
Author: petr
Status: does not work with debhelper,
	patch of source should be in xawtv-3.94/dist/,
	regenerate configure



diff -ur xawtv-3.94/debian/control xawtv-3.94/debian/control
--- xawtv-3.94/debian/control	2006-06-09 22:30:47.000000000 +0200
+++ xawtv-3.94/debian/control	2006-06-09 22:26:39.000000000 +0200
@@ -1,14 +1,14 @@
 Source: xawtv
 Section: graphics
 Priority: extra
-Build-Depends: libjpeg-dev, libncurses5-dev, libx11-dev, libxext-dev, libxinerama-dev, libxmu-dev, libxpm-dev, libxrandr-dev, libxt-dev, libxv-dev, libxxf86dga-dev, libxxf86vm-dev, x-dev, liblircclient-dev, debhelper (>= 4.1.16), xutils, xbase-clients, libaa1-dev, libasound2-dev, libzvbi-dev (>= 0.2.1), libquicktime-dev, libxaw7-dev, xlibmesa-gl-dev, libxft2-dev, po-debconf, libpng-dev, libglib1.2-dev, libfs-dev
+Build-Depends: libjpeg-dev, libncurses5-dev, libx11-dev, libxext-dev, libxinerama-dev, libxmu-dev, libxpm-dev, libxrandr-dev, libxt-dev, libxv-dev, libxxf86dga-dev, libxxf86vm-dev, x-dev, liblircclient-dev, debhelper (>= 4.1.16), xutils, xbase-clients, libaa1-dev, libasound2-dev [!kfreebsd-i386 !kfreebsd-amd64 !hurd-i386], libzvbi-dev (>= 0.2.1), libquicktime-dev, libxaw7-dev, xlibmesa-gl-dev, libxft2-dev, po-debconf, libpng-dev, libglib1.2-dev, libfs-dev
 Maintainer: Gerd Knorr <kraxel@debian.org>
 Standards-Version: 3.6.1
 
 Package: xawtv
 Section: x11
 Architecture: any
-Depends: ${shlibs:Depends}, v4l-conf, xutils, debconf, scantv, xawtv-plugins (=${Source-Version}), pia
+Depends: ${shlibs:Depends}, ${v4l-conf}, xutils, debconf, scantv, xawtv-plugins (=${Source-Version}), pia
 Suggests: xawtv-plugin-qt (=${Source-Version}), tv-fonts
 Description: X11 TV application
  TV application for X11.  Supports video4linux
@@ -23,7 +23,7 @@
  playback movie files which are recorded from TV.
 
 Package: fbtv
-Architecture: any
+Architecture: linux-any
 Depends: ${shlibs:Depends}, v4l-conf, xawtv-plugins (=${Source-Version})
 Suggests: xawtv
 Description: linux console (fbcon) TV application
@@ -59,14 +59,14 @@
 
 Package: radio
 Section: sound
-Architecture: any
+Architecture: linux-any
 Depends: ${shlibs:Depends}
 Description: ncurses-based radio application
  This is a ncurses-based radio application.  It
  supports the video4linux API.
 
 Package: v4l-conf
-Architecture: any
+Architecture: linux-any
 Depends: ${shlibs:Depends}
 Conflicts: suidmanager (<< 0.50)
 Description: tool to configure video4linux drivers
diff -ur xawtv-3.94/debian/rules xawtv-3.94/debian/rules
--- xawtv-3.94/debian/rules	2006-06-09 22:30:47.000000000 +0200
+++ xawtv-3.94/debian/rules	2006-06-09 22:24:50.000000000 +0200
@@ -31,7 +31,7 @@
 
 install: build-it
 	(cd $(WORK)/* && make DESTDIR=$(DEST) install)
-	dh_movefiles
+	dh_movefiles -s
 
 	# fix v4l-conf suid-root lintian warning
 	mkdir -p debian/v4l-conf/usr/share/lintian/overrides
@@ -63,7 +63,11 @@
 	dh_installdeb
 #	dh_perl
 	dh_shlibdeps
-	dh_gencontrol
+ifeq (linux,$(DEB_HOST_ARCH_OS))
+	dh_gencontrol -s -- -Vv4l-conf="v4l-conf"
+else
+	dh_gencontrol -s
+endif
 	dh_md5sums
 	dh_builddeb
 
diff -ur xawtv-3.94/debian/v4l-conf.files xawtv-3.94/debian/v4l-conf.files
--- xawtv-3.94/debian/v4l-conf.files	2006-06-09 22:30:47.000000000 +0200
+++ xawtv-3.94/debian/v4l-conf.files	2006-06-09 22:22:49.000000000 +0200
@@ -1,3 +1,5 @@
 usr/bin/v4l-conf
 usr/share/man/man8/v4l-conf.8
 usr/share/man/*/man8/v4l-conf.8
+usr/bin/v4l-info
+usr/share/man/man1/v4l-info.1
diff -ur xawtv-3.94/debian/xawtv.files xawtv-3.94/debian/xawtv.files
--- xawtv-3.94/debian/xawtv.files	2006-06-09 22:30:47.000000000 +0200
+++ xawtv-3.94/debian/xawtv.files	2006-06-09 22:22:15.000000000 +0200
@@ -4,7 +4,6 @@
 /usr/bin/rootv
 /usr/bin/ntsc-cc
 /usr/bin/subtitles
-/usr/bin/v4l-info
 /usr/share/man/man1/v4lctl.1
 /usr/share/man/*/man1/v4lctl.1
 /usr/share/man/man1/xawtv.1
@@ -16,7 +15,6 @@
 /usr/share/man/man1/ntsc-cc.1
 /usr/share/man/man1/subtitles.1
 /usr/share/man/*/man1/subtitles.1
-/usr/share/man/man1/v4l-info.1
 /usr/share/man/man5/xawtvrc.5
 /usr/share/man/*/man5/xawtvrc.5
 /etc/X11/app-defaults/Xawtv
diff -ur xawtv-3.94/work/xawtv-3.94/configure.ac xawtv-3.94/work/xawtv-3.94/configure.ac
--- xawtv-3.94/work/xawtv-3.94/configure.ac	2004-08-18 17:48:09.000000000 +0200
+++ xawtv-3.94/work/xawtv-3.94/configure.ac	2006-06-09 20:25:01.000000000 +0200
@@ -77,6 +77,9 @@
 		LDLIBS="$LDLIBS -L/usr/local/lib"
 		FOUND_OS="bsd"
 		;;
+	GNU/kFreeBSD)
+		FOUND_OS="bsd"
+		;;
 	*)
 		AC_MSG_CHECKING(if xawtv will build on `uname -s`)
 		AC_MSG_RESULT(maybe)
diff -ur xawtv-3.94/work/xawtv-3.94/libng/devices.c xawtv-3.94/work/xawtv-3.94/libng/devices.c
--- xawtv-3.94/work/xawtv-3.94/libng/devices.c	2004-03-25 11:35:36.000000000 +0100
+++ xawtv-3.94/work/xawtv-3.94/libng/devices.c	2006-06-09 19:37:12.000000000 +0200
@@ -9,7 +9,7 @@
 /*
  * default devices names
  */
-#if defined(__OpenBSD__) || defined(__FreeBSD__) || defined(__NetBSD__)
+#if defined(__OpenBSD__) || defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__NetBSD__)
 struct ng_device_config ng_dev = {
     video:  "/dev/bktr0",
     radio:  NULL,
diff -ur xawtv-3.94/work/xawtv-3.94/vbistuff/ntsc-cc.c xawtv-3.94/work/xawtv-3.94/vbistuff/ntsc-cc.c
--- xawtv-3.94/work/xawtv-3.94/vbistuff/ntsc-cc.c	2003-02-14 15:14:08.000000000 +0100
+++ xawtv-3.94/work/xawtv-3.94/vbistuff/ntsc-cc.c	2006-06-09 19:38:23.000000000 +0200
@@ -461,7 +461,7 @@
 	return 0;
 }
 
-static int RAW(int data)
+static int raw(int data)
 {
 	int b1, b2;
 	if (data == -1)
@@ -687,7 +687,7 @@
 				  x+1,128-buf[2048 * rawline+x*2+2]/2);
 		  }
 #endif
-		  RAW(decode(&buf[2048 * rawline]));
+		  raw(decode(&buf[2048 * rawline]));
 #ifndef X_DISPLAY_MISSING
 		  if (debugwin) {
 		    XFlush(dpy);
